src/libraries/Microsoft.PowerFx.Core/Functions/IAsyncTexlFunction4.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    // A Texl function capable of async invokes, using TimeZoneInfo and IRContext. 
+    internal interface IAsyncTexlFunction4
+    {
+        Task<FormulaValue> InvokeAsync(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args, CancellationToken cancellationToken);
```

Comment:
>IRContext [](http://example.com/codeflow?start=66&length=9)

IRContext was a mistake - can we avoid using here, and just pass the FormulaType?

Reply:
Not a mistake per say
I don't see a problem w/ it. Can you explain?

Reply:
Done